docker-java-transport-okhttp/src/main/java/com/github/dockerjava/okhttp/OkDockerHttpClient.java

```diff
@@ -0,0 +1,284 @@
+import com.github.dockerjava.core.SSLConfig;
+import okhttp3.ConnectionPool;
+import okhttp3.Dns;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import okio.BufferedSink;
+import okio.Okio;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.X509TrustManager;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public final class OkDockerHttpClient implements DockerHttpClient {
+
+    public static final class Factory {
+
+        private DockerClientConfig dockerClientConfig = null;
+
+        private Integer readTimeout = null;
+
+        private Integer connectTimeout = null;
+
+        private Boolean retryOnConnectionFailure = null;
+
+        public Factory dockerClientConfig(DockerClientConfig value) {
+            this.dockerClientConfig = value;
+            return this;
+        }
+
+        public Factory readTimeout(Integer value) {
+            this.readTimeout = value;
+            return this;
+        }
+
+        public Factory connectTimeout(Integer value) {
+            this.connectTimeout = value;
+            return this;
+        }
+
+        Factory retryOnConnectionFailure(Boolean value) {
+            this.retryOnConnectionFailure = value;
+            return this;
+        }
+
+        public OkDockerHttpClient build() {
+            return new OkDockerHttpClient(
+                dockerClientConfig,
+                readTimeout,
+                connectTimeout,
+                retryOnConnectionFailure
+            );
+        }
+    }
+
+    private static final String SOCKET_SUFFIX = ".socket";
+
+    final OkHttpClient client;
+
+    final OkHttpClient streamingClient;
+
+    private final HttpUrl baseUrl;
+
+    private OkDockerHttpClient(
+        DockerClientConfig dockerClientConfig,
+        Integer readTimeout,
+        Integer connectTimeout,
+        Boolean retryOnConnectionFailure
+    ) {
+        okhttp3.OkHttpClient.Builder clientBuilder = new okhttp3.OkHttpClient.Builder()
+            .addNetworkInterceptor(new HijackingInterceptor())
+            .readTimeout(0, TimeUnit.MILLISECONDS)
+            .retryOnConnectionFailure(true);
+
+        if (readTimeout != null) {
+            clientBuilder.readTimeout(readTimeout, TimeUnit.MILLISECONDS);
+        }
+
+        if (connectTimeout != null) {
+            clientBuilder.connectTimeout(connectTimeout, TimeUnit.MILLISECONDS);
+        }
+
+        if (retryOnConnectionFailure != null) {
+            clientBuilder.retryOnConnectionFailure(retryOnConnectionFailure);
+        }
+
+        URI dockerHost = dockerClientConfig.getDockerHost();
+        switch (dockerHost.getScheme()) {
```

Comment:
Why switch necessary if we have the fallthrough for unix?

Reply:
1) old copied code
2) "one case per line" IMO reads easier